### ISSUE TITLE
tslint broke when using tslint version < 4.0

### ISSUE THREAD

**Issue by iTouchTheSky** [Microsoft/vscode-tslint#257]
Given the following `tslint.json` : 
````json
{
  "semicolon": [true, "always"]
}
````

If I have the following code : 
````typescript
return someFunction.then(data => {
  return {
    someAttr: true
  }
})
````

The extension gives no errors, yet running a gulp lint task with the same `tsconfig.json` gives : 

> (semicolon) somefile.ts[4, 3]: Missing semicolon
> (semicolon) somefile.ts[5, 2]: Missing semicolon

**Comment by egamma** [comment]
I cannot reproduce this in my setup, see below. Is it just this rule or do you suspect that no linting is taking place?

Can you please send me a trace when the linting fails. To create a trace:
- Define the user setting `tslint.trace.server` : "verbose"
- Restart vscode
- Open the .ts file where linting fails
- Open the tslint output channel by clicking the TSLint item in the status line
- Attach the output to this issue

Thanks in advance

![image](https://user-images.githubusercontent.com/172399/30416238-b8e529bc-992b-11e7-87f9-c5854a3bacd3.png)

**Comment by iTouchTheSky** [comment]
Finally it seems like tslint doesn't work at all anymore with latest VSCode Insiders ...

What it looks like : 
![image](https://user-images.githubusercontent.com/2175877/30430147-45df4806-9928-11e7-920f-f86bd1280f99.png)

The trace : 
> [Trace - 08:37:23] Sending request 'initialize - (0)'.
Params: {
    "processId": 2888,
    "rootPath": "c:\\ico\\test_lint",
    "rootUri": "file:///c%3A/ico/test_lint",
    "capabilities": {
        "workspace": {
            "didChangeConfiguration": {
                "dynamicRegistration": true
            },
            "didChangeWatchedFiles": {
                "dynamicRegistration": true
            },
            "symbol": {
                "dynamicRegistration": true
            },
            "executeCommand": {
                "dynamicRegistration": true
            },
            "workspaceFolders": true,
            "configuration": true
        },
        "textDocument": {
            "synchronization": {
                "dynamicRegistration": true,
                "willSave": true,
                "willSaveWaitUntil": true,
                "didSave": true
            },
            "completion": {
                "dynamicRegistration": true,
                "completionItem": {
                    "snippetSupport": true
                }
            },
            "hover": {
                "dynamicRegistration": true
            },
            "signatureHelp": {
                "dynamicRegistration": true
            },
            "definition": {
                "dynamicRegistration": true
            },
            "references": {
                "dynamicRegistration": true
            },
            "documentHighlight": {
                "dynamicRegistration": true
            },
            "documentSymbol": {
                "dynamicRegistration": true
            },
            "codeAction": {
                "dynamicRegistration": true
            },
            "codeLens": {
                "dynamicRegistration": true
            },
            "formatting": {
                "dynamicRegistration": true
            },
            "rangeFormatting": {
                "dynamicRegistration": true
            },
            "onTypeFormatting": {
                "dynamicRegistration": true
            },
            "rename": {
                "dynamicRegistration": true
            },
            "documentLink": {
                "dynamicRegistration": true
            }
        }
    },
    "trace": "verbose",
    "workspaceFolders": [
        {
            "uri": "file:///c%3A/ico/test_lint",
            "name": "test_lint"
        }
    ]
}


[Trace - 08:37:24] Received response 'initialize - (0)' in 1009ms.
Result: {
    "capabilities": {
        "textDocumentSync": 1,
        "codeActionProvider": true
    }
}


[Info  - 08:37:24] Linter is running.
[Trace - 08:37:24] Sending notification 'initialized'.
Params: {}


[Trace - 08:37:24] Sending notification 'workspace/didChangeConfiguration'.
Params: {
    "settings": {
        "tslint": {
            "enable": true,
            "jsEnable": false,
            "rulesDirectory": "",
            "validateWithDefaultConfig": false,
            "configFile": "",
            "ignoreDefinitionFiles": true,
            "exclude": "",
            "run": "onType",
            "nodePath": "",
            "autoFixOnSave": false,
            "alwaysShowRuleFailuresAsWarnings": false,
            "trace": {
                "server": "verbose"
            }
        }
    }
}


[Trace - 08:37:24] Sending notification 'textDocument/didOpen'.
Params: {
    "textDocument": {
        "uri": "file:///c%3A/ico/test_lint/test.ts",
        "languageId": "typescript",
        "version": 1,
        "text": "let test = true\r\nlet t = {\r\n    something: true\r\n}\r\n\r\nconst someFunction = (callback: () => void) => {\r\n    callback();\r\n}\r\n\r\nsomeFunction(() => {\r\n\r\n})\r\n"
    }
}


[Trace - 08:37:24] Sending request 'textDocument/codeAction - (1)'.
Params: {
    "textDocument": {
        "uri": "file:///c%3A/ico/test_lint/test.ts"
    },
    "range": {
        "start": {
            "line": 0,
            "character": 15
        },
        "end": {
            "line": 0,
            "character": 15
        }
    },
    "context": {
        "diagnostics": []
    }
}


[Trace - 08:37:24] Received request 'workspace/configuration - (0)'.
Params: {
    "items": [
        {
            "scopeUri": "file:///c%3A/ico/test_lint/test.ts",
            "section": "tslint"
        }
    ]
}


[Trace - 08:37:24] Sending response 'workspace/configuration - (0)'. Processing request took 1ms
[Trace - 08:37:24] Received response 'textDocument/codeAction - (1)' in 2ms.
Result: []


[Trace - 08:37:24] NODE_PATH value is: C:\Users\maximel\AppData\Roaming\npm\node_modules;C:\Users\maximel\AppData\Roaming\npm\node_modules
[Trace - 08:37:25] Received notification 'window/logMessage'.
Params: {
    "type": 3,
    "message": "TSLint library loaded from: C:\\Users\\maximel\\AppData\\Roaming\\npm\\node_modules\\tslint\\lib\\tslint.js"
}


[Info  - 08:37:25] TSLint library loaded from: C:\Users\maximel\AppData\Roaming\npm\node_modules\tslint\lib\tslint.js
[Trace - 08:37:25] tslint configuration:
{ rules: 
   { indent: [ true, 'spaces' ],
     'member-access': [ true ],
     'member-ordering': [ true ],
     'no-consecutive-blank-lines': true,
     'no-console': [ true, 'debug', 'log', 'info', 'warn', 'table' ],
     'no-duplicate-key': true,
     'no-duplicate-variable': true,
     'no-eval': true,
     'no-trailing-whitespace': true,
     'no-use-before-declare': true,
     'no-var-keyword': true,
     'one-line': [ true, 'check-open-brace', 'check-catch', 'check-else' ],
     quotemark: [ true, 'single', 'avoid-escape' ],
     radix: true,
     semicolon: [ true, 'always' ],
     'triple-equals': [ true, 'allow-undefined-check' ],
     'variable-name': 
      [ true,
        'check-format',
        'allow-leading-underscore',
        'ban-keywords' ],
     whitespace: [ true ],
     'no-debugger': true },
  rulesDirectory: [],
  extends: [] }
[Trace - 08:37:25] Linting: with tslint < version 4
[Trace - 08:37:25] Linting: ended linting
[Trace - 08:37:25] Received notification 'tslint/status'.
Params: {
    "state": 1
}


[Trace - 08:37:25] Received notification 'textDocument/publishDiagnostics'.
Params: {
    "uri": "file:///c%3A/ico/test_lint/test.ts",
    "diagnostics": []
}

**Comment by egamma** [comment]
@iTouchTheSky thanks for following with the trace. 

>[Trace - 08:37:25] Linting: with tslint < version 4
[Trace - 08:37:25] Linting: ended linting

Interestingly you are using a version that is older than 4.0 and I suspect I broke something in supporting this version. The API is different and there is another code path.

I will investigate into running with a tslint version that is older than 4.0. As a work around can you try to run your test with the latest tslint version?

**Comment by iTouchTheSky** [comment]
@egamma It seems to be working with the latest version, I will try to check the impacts of upgrading `tslint` in our end (since going from 3.14.0 to 5.7.0 should be breaking some stuff)

**Comment by egamma** [comment]
This should be fixable I'll look into this as soon as possible.

With regard to upgrading to 5.7, you will run into this: https://github.com/Microsoft/vscode-tslint/tree/master/tslint#faq. A middle step would be to upgrade to version 4.0.

**Comment by egamma** [comment]
@iTouchTheSky fixed and published version 1.0.6 with the fix.
